### PR TITLE
feat: Configure external devfile registries urls

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
- `status.DevfileRegistryURL` contain only url to registry deployed by operator (not external ones)
- `CHE_WORKSPACE_DEVFILE__REGISTRY__URL` contains all urls including external ones

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19781

### How to test this PR?

1. Deploy Eclipse Che
2. Configure `spec.server.externalDevfilerRegistries`
3. Check `CHE_WORKSPACE_DEVFILE__REGISTRY__URL` inside `che` config map. It must contain internal + external devfile registries urls

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [x] [Nightly OLM bundle is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by mmorhun — APPROVED** [review]

File: deploy/crds/org_v1_che_crd-v1beta1.yaml [hunk 1]
```diff
@@ -601,17 +601,28 @@ spec:
...
-                    Operator.
+                    ready-to-use devfiles. This will be automatically calculated by
+                    the Operator. See the `externalDevfileRegistry` and `externalDevfileRegistries`
+                    fields.
                   type: string
+                externalDevfileRegistries:
+                  description: External devfile registries, that serves sample, ready-to-use
+                    devfiles. See the `externalDevfileRegistry` field.
+                  items:
+                    description: Settings forconfiguration of the external devfile
```
Comment (sleshchenko) [inline deploy/crds/org_v1_che_crd-v1beta1.yaml:612]:
```suggestion
                    description: Settings for configuration of the external devfile

```

File: deploy/crds/org_v1_che_crd-v1beta1.yaml [hunk 2]
```diff
@@ -601,17 +601,28 @@ spec:
...
+                    properties:
+                      url:
+                        description: Public URL of the devfile registry.
+                        type: string
+                    type: object
+                  type: array
                 externalDevfileRegistry:
                   description: Instructs the Operator on whether to deploy a dedicated
                     devfile registry server. By default, a dedicated devfile registry
                     server is started. When `externalDevfileRegistry` is `true`, no
```
Comment (sleshchenko) [inline deploy/crds/org_v1_che_crd-v1beta1.yaml:623]: Is it true? I thought the purpose of PR was that add an ability to use external registry when internal (managed by operator) is still present.

**Pull request merged**